verticapy/tests_new/machine_learning/vertica/__init__.py

```diff
@@ -17,48 +17,497 @@
 REL_TOLERANCE = 1e-6
 ABS_TOLERANCE = 1e-12
 
-rel_tolerance_map = {
-    "RandomForestRegressor": 1e-0,
-    "RandomForestClassifier": 1e-0,
-    "DecisionTreeRegressor": 1e-0,
-    "DecisionTreeClassifier": 1e-0,
-    "DummyTreeRegressor": 1e-01,
-    "DummyTreeClassifier": 1e-01,
-    "XGBRegressor": 1e-0,
-    "XGBClassifier": 1e-0,
-    "Ridge": 1e-02,
-    "Lasso": 1e-02,
-    "ElasticNet": 1e-02,
-    "LinearRegression": 1e-02,
-    "LinearSVR": 1e-01,
-    "PoissonRegressor": 1e-01,
-    "AR": 1e-01,
-    "MA": 1e-00,
-    "ARMA": 1e-01,
-    "ARIMA": 1e-00,
-    "TENSORFLOW": 1e-06,
-    "TFIDF": 1e-01,
-}
-
-abs_tolerance_map = {
-    "RandomForestRegressor": 1e-0,
-    "RandomForestClassifier": 1e-0,
-    "DecisionTreeRegressor": 1e-0,
-    "DecisionTreeClassifier": 1e-0,
-    "DummyTreeRegressor": 1e-01,
-    "DummyTreeClassifier": 1e-01,
-    "XGBRegressor": 1e-0,
-    "XGBClassifier": 1e-0,
-    "Ridge": 1e-02,
-    "Lasso": 1e-02,
-    "ElasticNet": 1e-02,
-    "LinearRegression": 1e-02,
-    "LinearSVR": 1e-01,
-    "PoissonRegressor": 1e-01,
-    "AR": 1e-01,
-    "MA": 1e-01,
-    "ARMA": 1e-02,
-    "ARIMA": 1e-00,
-    "TENSORFLOW": 1e-06,
+rel_abs_tol_map = {
+    "RandomForestRegressor": {
+        "explained_variance": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "max_error": {"rel": 1e-00, "abs": ABS_TOLERANCE},
+        "median_absolute_error": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "mean_absolute_error": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "mean_squared_error": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "mean_squared_log_error": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        "rmse": {"rel": 1e-01, "abs": ABS_TOLERANCE},
+        **dict.fromkeys(["r2", "R-squared"], {"rel": 1e-01, "abs": ABS_TOLERANCE}),
```

Comment:
I think we should be as tight as possible on these R-squared values. Did you try 0.05? I think with the same data, we should be in that range.

Reply:
As discussed, I have added code to calculate tolerance using
_rel_tol = abs(vpy_score - py_score) / min(abs(vpy_score), abs(py_score))
_abs_tol = abs(vpy_score - py_score) / (1 + min(abs(vpy_score), abs(py_score)))
and accordingly updated value in __init__.py file.